zebra-consensus/src/transaction.rs

```diff
@@ -182,14 +201,32 @@ where
     ZS: Service<zs::Request, Response = zs::Response, Error = BoxError> + Send + Clone + 'static,
     ZS::Future: Send + 'static,
 {
+    /// Verify a V4 transaction.
+    ///
+    /// Returns a set of asynchronous checks that must all succeed for the transaction to be
+    /// considered valid. These checks include:
+    ///
+    /// - transparent transfers
+    /// - sprout shielded data
+    /// - sapling shielded data
+    ///
+    /// The parameters of this method are:
+    ///
+    /// - the `request` to verify (that contains the transaction and other metadata, see [`Request`]
+    ///   for more information)
+    /// - the `network` to consider when verifying
+    /// - the `script_verifier` to use for verifying the transparent transfers
+    /// - the transparent `inputs` in the transaction
+    /// - the Sprout `joinsplit_data` shielded data in the transaction
+    /// - the `sapling_shielded_data` in the transaction
     async fn verify_v4_transaction(
```

Comment:
A nit: this is an `async` function that returns `Result<AsyncChecks, _>`, vs `verify_v5_transaction`, which here is now !`async`, but now returns  `Result<AsyncChecks, _>` as well. We should be consistent either way, should the latter still be `async`, or the former !`async`?

Reply:
That's a good point. I might be able to make `verify_v4_transaction` non-async as well, but more refactoring is needed in future PRs. I've made `verify_v5_transaction` async again, and if I manage to make both of them non-async later I'll remove the `async` in a single commit :+1: